go/vt/sqlparser/ast.go

```diff
@@ -325,7 +325,7 @@ type (
 		Action     InsertAction
 		Comments   *ParsedComments
 		Ignore     Ignore
-		Table      TableName
+		Table      *AliasedTableExpr
```

Comment:
I think we should add a couple of tests to parse_test that shows this change

Reply:
This is only done for semantic analyzer to work. otherwise at the syntax level this will never happen.
For the same reason we are using a function to convert TableName to *AliasedTableExpr but not changing the sql.y file at the syntax level.